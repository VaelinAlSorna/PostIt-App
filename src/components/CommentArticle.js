import React from 'react';
import uuid from 'uuid';
import moment from 'moment';

class CommentArticle extends React.Component {
    state = {
        comment: '',
        createdAt: moment().format('D. M. Y'),
        error: ''
    };
    onTextareaChange = (e) => {
        const comment = e.target.value;
        this.setState(() => ({
            comment
        }));
    };
    onSubmit = (e) => {
        e.preventDefault();
        if (this.state.comment.trim().length < 3) {
            this.setState(() => ({
                error: 'Yout comment must contain at least 3 characters.'
            }));
        } else {
            const id = uuid();
            this.props.onSubmit({...this.state, id});
            this.setState(() =>  ({
                comment: '',
                error: ''
            }));
        }
    }
    render () {
        return (
            <form onSubmit={this.onSubmit}>
                { this.state.error && <p>{this.state.error}</p>}
                <textarea 
                    onChange={this.onTextareaChange} 
                    placeholder="What are your thoughts ?"
                    value={this.state.comment}
                >
                </textarea>
                <input type="submit" value="Comment" />
            </form>
        );
    };
};

export default CommentArticle;