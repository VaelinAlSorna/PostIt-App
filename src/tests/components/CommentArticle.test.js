import React from 'react';
import { shallow } from 'enzyme';
import CommentArticle from '../../components/CommentArticle';

test('should render CommentArticle with default data', () => {
    const wrapper = shallow(<CommentArticle likeOrCommentAllowed={false} />);
    expect(wrapper).toMatchSnapshot();
});

test('should print error when Submiting with invalid data', () => {
    const wrapper = shallow(<CommentArticle likeOrCommentAllowed={false} />);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(wrapper.state('error')).toBe('Yout comment must contain at least 3 characters.')
});

test('should call onSubmit when Submiting with correct data', () => {
    const onSubmitSpy = jest.fn();
    const value = 'test';
    const wrapper = shallow(<CommentArticle likeOrCommentAllowed={false} onSubmit={onSubmitSpy}/>);
    wrapper.find('textarea').simulate('change', {
        target: { value }
    });
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        comment: value,
        createdAt: '1.1.1970'
    })
});

test('should change comment value when changing textarea', () => {
    const value = 'test';
    const wrapper = shallow(<CommentArticle likeOrCommentAllowed={false} />);
    wrapper.find('textarea').simulate('change', {
        target: { value }
    });
    expect(wrapper.state('comment')).toBe(value);
});

test('comment button should be disabled if likeOrCommentAllowed props is false ', () => {
    const wrapper = shallow(<CommentArticle likeOrCommentAllowed={true} />);
    expect(wrapper.find('input').props().disabled).toBe(true);
});




