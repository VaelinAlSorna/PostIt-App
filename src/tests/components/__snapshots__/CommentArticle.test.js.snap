// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CommentArticle with default data 1`] = `
<form
  onSubmit={[Function]}
>
  <textarea
    onChange={[Function]}
    placeholder="What are your thoughts ?"
    value=""
  />
  <input
    disabled={false}
    type="submit"
    value="Comment"
  />
</form>
`;
